### PR TITLE
Perf improvements

### PR DESCRIPTION
**Pull request by franz1981**
@jabolina @belaban While playing with the agent I've spotted these low hanging fruits: no idea if they worth an issue but I can add more details here in case

### CODE REVIEW

File: src/org/jgroups/protocols/raft/LogEntries.java [hunk 1]
```diff
@@ -63,7 +65,15 @@ public LogEntry[] toArray() {
     }
 
     public long totalSize() {
-        return entries == null? 0 : entries.stream().filter(Objects::nonNull).map(e -> e.length).reduce(0, Integer::sum);
+        final ArrayList<LogEntry> entries = this.entries;
```
Comment (jabolina) [inline src/org/jgroups/protocols/raft/LogEntries.java:68]: Missing the null check for `entries`.

**Review by jabolina — COMMENTED** [review]
I was [also experimenting](https://github.com/belaban/jgroups-raft/compare/master...jabolina:jgroups-raft:allocation-request-table) at the `RequestTable` but didn't get meaningful results in terms of reducing allocations or GC pressure when profiling CPU :disappointed:

**Comment by belaban** [comment]
Hey Francesco
do you have proof that the stream API is slower than the iteration?

**Comment by franz1981** [comment]
> Hey Francesco
do you have proof that the stream API is slower than the iteration?

yep, allocation-wise by collecting `-e alloc` on our perf tests it popup as one of hot allocation code paths; and perf-wise allocations that can be saved, often bring some speedup.
I can collect some data if required (I've deleted them few days ago)

**Comment by belaban** [comment]
No, I trust your experience. The problem is that you're making me go back to C-style code and forfeit all the nice features of Java 8+ (ie., streams) :-)

**Comment by franz1981** [comment]
I know @belaban I've already raised this to the JDK team (Oracle, not ours) many times: Stream right now are not suitable on the hot path with OpenJDK (but GraalVm does a way better job there AFAIK)

**Pull request merged**